mm2src/coins/eth.rs

```diff
@@ -249,6 +249,29 @@ pub mod gas_limit {
     pub const ETH_MAX_TRADE_GAS: u64 = 150_000;
 }
 
+/// Coin conf param to override default gas limits
+#[derive(Default, Deserialize)]
```

Comment:
Question/suggestion

Instead of using `derive(Default)` and unwrapping gas limit fields every time with compile-time constants, don't you think it would be better to implement the `Default` trait ourselves? We can set the values directly from the `mod gas_limit`, which would allow us to provide the EthGasLimit type with non-optional gas limit fields.
![Screenshot 2024-06-09 at 13 14 42](https://github.com/KomodoPlatform/komodo-defi-framework/assets/52405288/34f5e78a-bffe-4f62-bf80-0863a9bb28fd)

Reply:
Agreed!

I guess this is what you wanted to have:

```rust
/// Coin conf param to override default gas limits
#[derive(Deserialize)]
#[serde(default)]
pub struct EthGasLimitOverride {
    /// Gas limit for sending coins override value
    pub eth_send_coins: u64,
    /// Gas limit for sending ERC20 tokens override value
    pub eth_send_erc20: u64,
    /// Gas limit for swap payment tx with coins override value
    pub eth_payment: u64,
    /// Gas limit for swap payment tx with ERC20 tokens override value
    pub erc20_payment: u64,
    /// Gas limit for swap receiver spend tx with coins override value
    pub eth_receiver_spend: u64,
    /// Gas limit for swap receiver spend tx with ERC20 tokens override value
    pub erc20_receiver_spend: u64,
    /// Gas limit for swap refund tx with coins override value
    pub eth_sender_refund: u64,
    /// Gas limit for swap refund tx with with ERC20 tokens override value
    pub erc20_sender_refund: u64,
    /// Gas limit for other operations override value
    pub eth_max_trade_gas: u64,
}

impl Default for EthGasLimitOverride {
    fn default() -> EthGasLimitOverride {
        EthGasLimitOverride {
            eth_send_coins: gas_limit::ETH_SEND_COINS,
            eth_send_erc20: gas_limit::ETH_SEND_ERC20,
            eth_payment: gas_limit::ETH_PAYMENT,
            erc20_payment: gas_limit::ERC20_PAYMENT,
            eth_receiver_spend: gas_limit::ETH_RECEIVER_SPEND,
            erc20_receiver_spend: gas_limit::ERC20_RECEIVER_SPEND,
            eth_sender_refund: gas_limit::ETH_SENDER_REFUND,
            erc20_sender_refund: gas_limit::ERC20_SENDER_REFUND,
            eth_max_trade_gas: gas_limit::ETH_MAX_TRADE_GAS,
        }
    }
}
```

Reply:
> I guess this is what you wanted to have

yes that what I proposed to implement. but then

@dimxy I think I got how to keep `EthGasLimitOverride` with non-optional fields and be able to build this structure from json safely in the case if `"gas_limit"` misssing some fields in coin config.

We should refactor this line `json::from_value(coin_conf["gas_limit"].clone()).map_err(|e| e.to_string())` here 
```rust
fn extract_gas_limit_from_conf(coin_conf: &Json) -> Result<EthGasLimitOverride, String> {
    if coin_conf["gas_limit"].is_null() {
        Ok(Default::default())
    } else {
        json::from_value(coin_conf["gas_limit"].clone()).map_err(|e| e.to_string())
    }
}
```
please parse from json `EthGasLimitOverrideWrap` structure, which is supposed to have optional fields. then implement `From` for `EthGasLimitOverride` structure (non-optional fields) where you can use `unwrap_or(gas_limit::ETH_SEND_COINS)` etc to unwrap None and Some and provide `EthGasLimitOverride` with non-optional fields. 

So you dont have to do this all the time in the code:
```rust
            EthCoinType::Eth => self.sign_and_send_transaction(
                value,
                Action::Call(address),
                vec![],
                U256::from(self.gas_limit.eth_send_coins.unwrap_or(gas_limit::ETH_SEND_COINS)),
            ),
```
Big number of such manual unwraps could lead to issues, bcz you and reviewers have to double check every time what you put in `unwrap_or` everywhere in the code

Reply:
thanks, fixed in [36f0601](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2137/commits/36f0601448064c18e1e8b30d720967323d2ac13d) using #[serde(default)]

Reply:
> thanks, fixed in [36f0601](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2137/commits/36f0601448064c18e1e8b30d720967323d2ac13d) using #[serde(default)]

Oh, cool solution